samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -176,8 +181,93 @@ static void on_location_update(const struct location_event_data * const location
 	}
 }
 
+/**
+ * @brief Receives general device messages from nRF Cloud, checks if they are AT command requests,
+ * and performs them if so, transmitting the modem response back to nRF Cloud.
+ *
+ * Try sending {"appId":"MODEM", "messageType":"CMD", "data":"AT+CGMR"}
+ * in the nRF Cloud Portal Terminal card.
+ *
+ * @param msg - The device message to check.
+ */
+static char at_req_resp_buf[CONFIG_AT_HOST_CMD_MAX_LEN];
+static void handle_at_cmd_requests(const char * const msg) {
+
+	/* Attempt to parse the message as if it is JSON */
+	struct cJSON *msg_obj = cJSON_Parse(msg);
+
+	if (!msg_obj) {
+		/* The message isn't JSON or otherwise couldn't be parsed. */
+		return;
+	}
+
+	/* Check whether the JSON device message is a modem AT command */
+	if (strcmp(cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, NRF_CLOUD_JSON_APPID_KEY)),
+		   NRF_CLOUD_JSON_APPID_VAL_MODEM) != 0) {
+		goto cleanup;
+	}
+	if (strcmp(cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, NRF_CLOUD_JSON_MSG_TYPE_KEY)),
+		   NRF_CLOUD_JSON_MSG_TYPE_VAL_CMD) != 0) {
+		goto cleanup;
+	}
+
+	/* If it is, extract the requested command string */
+	const char *cmd = cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, "data"));
+
+	LOG_DBG("Modem AT command requested: %s", cmd);
+
+	/* Execute the command and store the result */
+	int err = nrf_modem_at_cmd(at_req_resp_buf, sizeof(at_req_resp_buf), "%s", cmd);
+
+	/* Trim \r\n from modem response */
```

Comment:
why modify the response?

Reply:
Since the response is being wrapped in a JSON object, the training \r\n are useless and a bit ugly in the dashboard IMO

Reply:
useless unless you are passing the result to some AT command parser which might expect them... but that might be getting a bit into the weeds.  maybe update the comment to say it is trimmed for improved readability in the nrf cloud terminal card?

Reply:
Sure thing, I'll update the comment.